src/packaging/tags.py

```diff
@@ -473,7 +473,7 @@ def _linux_platforms(is_32bit: bool = _32_BIT_INTERPRETER) -> Iterator[str]:
         if linux == "linux_x86_64":
             linux = "linux_i686"
         elif linux == "linux_aarch64":
-            linux = "linux_armv7l"
+            linux = "linux_armv8l"
```

Comment:
This one changes the default behavior for 32-bit python on aarch64 systems.
This is to ensure that reported tags are the same if you run python directly or under linux32 emulation (which will report `linux_armv8l`).

Reply:
Can armv7l run arm8l binaries in all cases? I'm assuming not, so I'm concerned about this misreporting?

Reply:
I'm not sure I understand the question here.
If we're running an aarch64 kernel, we are, by definition, on an armv8 chip or newer so we can't be misreporting by saying `linux_armv8l`, we'll never hit this line on an armv7 chip.

Reply:
I guess I don't understand why this is a good change when this doesn't work for arm7l machines?

Reply:
This PR doesn't change a thing for armv7l machines. They will still be reporting the same thing as they did before. It will only change things for armv8+ machines running a 32-bit python.

The following extract which is a bit longer than the one line diff might help going through what happens.

```
    if is_32bit:
        if linux == "linux_x86_64":
            linux = "linux_i686"
        elif linux == "linux_aarch64":
            linux = "linux_armv8l"
```

armv7l machines are 32-bit only (https://en.wikipedia.org/wiki/ARM_architecture_family#32-bit_architecture).
The kernel will thus never report  an "aarch64" machine which is the one reported for 64-bit arm.
So, in the case of armv7l, we were not overriding the `linux` and this is still the case.